artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -742,6 +756,25 @@ public void addConsumer(ConsumerInfo info) throws Exception {
       }
    }
 
+   public void setConnectionEntry(ConnectionEntry connectionEntry) {
+      this.connectionEntry = connectionEntry;
+   }
+
+   public void setUpTtl(final long inactivityDuration, final long inactivityDurationInitialDelay, final boolean useKeepAlive) {
+      this.useKeepAlive = useKeepAlive;
+      this.maxInactivityDuration = inactivityDuration;
+
+      protocolManager.getScheduledPool().schedule(new Runnable() {
```

Comment:
With Openwire client there is only one maxInactivityDuration, so the client->server and server->client should be the same (if the server and client are different values, the smaller is taken). So in case of 1 seconds the server will send a ping at leas for every half of the seconds in order to avoid client timing out the connection.